src/Catalog/Dnx/DnxCatalogCollector.cs

```diff
@@ -34,24 +36,33 @@ public DnxCatalogCollector(Uri index, StorageFactory storageFactory, Func<HttpMe
                 string id = item["nuget:id"].ToString().ToLowerInvariant();
                 string version = NuGetVersionUtility.NormalizeVersion(item["nuget:version"].ToString().ToLowerInvariant());
                 string type = item["@type"].ToString().Replace("nuget:", Schema.Prefixes.NuGet);
+                var destinationStorage = _storageFactory.Create(id);
 
                 if (type == Schema.DataTypes.PackageDetails.ToString())
                 {
-                    // Add/update package
-                    string nuspec = await LoadNuspec(client, id, version, cancellationToken);
-                    if (nuspec != null)
+                    var sourceUri = new Uri(ContentBaseAddress, string.Format("{0}.{1}.nupkg", id, version));
+                    var destinationUri = destinationStorage.GetUri(DnxMaker.GetRelativeAddressNupkg(id, version));
+                    if (!await destinationStorage.AreSyncronized(sourceUri, destinationUri))
                     {
-                        var requestUri = Utilities.GetNugetCacheBustingUri(new Uri(ContentBaseAddress, string.Format("{0}.{1}.nupkg", id, version)));
-                        using (Stream stream = await client.GetStreamAsync(requestUri))
+                        // Add/update package
+                        string nuspec = await LoadNuspec(client, id, version, cancellationToken);
```

Comment:
Can we combine the `LoadNuspec` and `GetStreamAsync` download? If we save to a temp file on disk, then we should only need to download once.

Reply:
This changes will introduce more risk that I propose to not take in this hotfix. I opened a task for future improvements. I added this suggestion to the task/issue.